libraries/cmake/source/lief/CMakeLists.txt

```diff
@@ -0,0 +1,236 @@
+# Copyright (c) 2014-present, The osquery authors
+#
+# This source code is licensed as defined by the LICENSE file found in the
+# root directory of this source tree.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+
+function(lief)
+  set(library_root "${CMAKE_CURRENT_SOURCE_DIR}/src")
+  set(LIEF_ELF_SUPPORT 0)
+  set(LIEF_OAT_SUPPORT 0)
+  set(LIEF_DEX_SUPPORT 0)
+  set(LIEF_VDEX_SUPPORT 0)
+  set(LIEF_ART_SUPPORT 0)
+  set(LIEF_JSON_SUPPORT 0)
+  set(LIEF_LOGGING_SUPPORT 0)
+  set(LIEF_LOGGING_DEBUG_SUPPORT 0)
+  set(LIEF_FROZEN_ENABLED 0)
+  set(LIEF_PE_SUPPORT 0)
+  set(LIEF_MACHO_SUPPORT 0)
```

Comment:
If these are set for the `configure_file` call to generate the correct files, I wouldn't do it like this, because you're still generating a configuration at runtime, while we would like to have a snapshot of it.

Run the lief configuration separately and then take the generated files and commit/use them on osquery like they are.

Reply:
i see that makes more sense. Removed